src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.CSharp.SourceGenerators.Testing.MSTest/IncrementalGeneratorVerifier.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.CodeAnalysis.CSharp.Testing.MSTest
+{
+    public static class IncrementalGeneratorVerifier
+    {
+        public static IncrementalGeneratorVerifier<TIncrementalGenerator> Create<TIncrementalGenerator>()
```

Comment:
No need to make new types, one can add ``, IIncrementalGenerator, new()`` to the contraints on the ones already accepting an ``ISourceGenerator`` that I see so far.

Reply:
```diff
public class CSharpSourceGeneratorTest<TSourceGenerator, TVerifier> : SourceGeneratorTest<TVerifier>
-         where TSourceGenerator : ISourceGenerator, new()
+         where TSourceGenerator : ISourceGenerator, IIncrementalGenerator, new()
        where TVerifier : IVerifier, new()
```

In SourceGeneratorTest that is inherited from this change:
- ``GetSourceGenerators`` to return ``object`` instead of ISourceGenerator so that way CSharpSourceGeneratorTest would just work with it (have it down cast to object) then have SourceGeneratorTest do a type check then cast operation to ISourceGenerator or to ``IIncrementalGenerator``.
- ``CreateGeneratorDriver`` to be defined as ``protected abstract GeneratorDriver CreateGeneratorDriver(Project project, ImmutableArray<object> sourceGenerators);`` (again when using it would need to type check and cast to use the instances).
- ``ApplySourceGeneratorAsync`` to be defined as ``private async Task<(Project project, ImmutableArray<Diagnostic> diagnostics)> ApplySourceGeneratorAsync(ImmutableArray<object> sourceGenerators, Project project, IVerifier verifier, CancellationToken cancellationToken)``
- ``VerifySourceGeneratorAsync`` to be defined as ``private async Task<ImmutableArray<Diagnostic>> VerifySourceGeneratorAsync(string language, ImmutableArray<object> sourceGenerators, SolutionState testState, Func<ImmutableArray<ISourceGenerator>, Project, IVerifier, CancellationToken, Task<(Project project, ImmutableArray<Diagnostic> diagnostics)>> getFixedProject, IVerifier verifier, CancellationToken cancellationToken)``